186.py

```diff
@@ -0,0 +1,27 @@
+class Solution:
+    def reverseWords(self, s: List[str]) -> None:
+        """
+        Do not return anything, modify s in-place instead.
+        """
+        def reverse(start: int, end: int) -> None:
+            while start < end:
+                s[start], s[end] = s[end], s[start]
+                start += 1
+                end -= 1
+
+        N = len(s)
+        reverse(0, N - 1)
+        start, end = 0, 0
+
+        while end < N:
```

Comment:
The logic inside your `while` loop is a bit complicated. All it needs to do are: 

1. For every `start`, find the next whitespace or the end of the string.
2. Reverse all the elements between the current `start` and the next whitespace.
3. Set `start` to the next word.
4. Repeat.

```python
def reverseWords(self, s: List[str]) -> None:
    """
    Do not return anything, modify s in-place instead.
    """
    def reverse(arr, l, r):
        while l < r:
            arr[l], arr[r] = arr[r], arr[l]
            l += 1
            r -= 1
            
    N = len(s)
    reverse(s, 0, N - 1)
    p = 0
    
    while p < N:
        q = p
        
        while q < N and s[q].isalpha():
            q += 1
            
        reverse(s, p, q - 1)
        
        p = q + 1